core/controllers/profile.py

```diff
@@ -77,6 +77,12 @@ def get(self, username):
         if not user_settings:
             raise self.PageNotFoundException
 
+        if user_settings.first_contribution_datetime:
```

Comment:
I don't think you need the if clause. Just bundle this in self.values.update below -- it's OK to pass None to the frontend. At least the keys are always the same.

You probably want to do something like:

```
    'first_contribution_datetime': (
        utils.get_time_in_millisecs(user_settings.first_contribution_datetime)
        if user_settings.first_contribution_datetime else None),
```